packages/core/src/Serialization/PlutusData/PlutusMap.ts

```diff
@@ -88,7 +88,13 @@ export class PlutusMap {
    * if there is no element with the given key.
    */
   get(key: PlutusData): PlutusData | undefined {
-    return this._map.get(key);
+    if (!this._map) return undefined;
+
+    const element = [...this._map.entries()].find((entry) => deepEquals(entry[0], key));
```

Comment:
Just a nitpick suggestion, entries is fine too
```suggestion
    const element = [...this._map.keys()].find((mapKey) => deepEquals(mapKey, key));
```